R/download_data_wrds_compustat.R

```diff
@@ -85,9 +89,44 @@ download_data_wrds_compustat <- function(type, start_date, end_date, additional_
         inv = at / at_lag - 1,
         inv = if_else(at_lag <= 0, NA, inv)
       )
+  }
+
+  if (grepl("compustat_quarterly", type, fixed = TRUE)) {
+    fundq_db <- tbl(con, in_schema("comp", "fundq"))
+
+    compustat <- fundq_db |>
+      filter(
+        indfmt == "INDL" &
+          datafmt == "STD" &
+          consol == "C" &
+          between(datadate, start_date, end_date)
+      ) |>
+      select(
+        gvkey, datadate, rdq, fqtr, fyearq,
+        all_of(additional_columns)
+      ) |>
+      collect()
 
     disconnection_connection(con)
 
-    processed_data
+    compustat <- compustat |>
+      drop_na(fqtr)|>
+      mutate(date = ceiling_date(datadate, "quarter") %m-% months(1),
+             timepoint = paste0(fyearq, fqtr)) |>
+      group_by(gvkey, timepoint) |>
+      filter(datadate == max(datadate)) |>
+      ungroup() |>
+      filter(if_else(is.na(rdq), TRUE, date < rdq)) |>
+      arrange(gvkey, datadate) |>
```

Comment:
The current logic just keeps gvkey-date combinations with exactly one row. The arrange statement does nothing imho. There should be only one row per gvkey-datadate combination. Can you explain what exactly needs to be dropped here?

Reply:
There might be several datadates that match the last. We must only have one. 

I now suggest a `slice_head()`, but I am happy for alternative suggestions.